src/mongo/delegates/Views.class.php

```diff
@@ -396,58 +406,16 @@ public function generateView($viewId,$resource=null,$context=null,$queueName=nul
             $t = new \Tripod\Timer();
             $t->start();
 
-            $from = $this->getFromCollectionForViewSpec($viewSpec);
+            $from = $viewSpec['from'];
             $collection = $this->config->getCollectionForView($this->storeName, $viewId);
 
+            $this->ensureIndices($viewSpec, $collection);
```

Comment:
This has come up before.  The issue is that it's not entirely clear how you would ensure that the indexes have been created.  That is, what process would exist to make sure this happens and, if it hasn't, does it?

As it is, `ensureIndex` isn't terribly expensive, so the feeling is that it's less expensive to ensureIndexes rather than miss them.